kotlin-result/build.gradle.kts

```diff
@@ -3,21 +3,6 @@ description = "A Result monad for modelling success or failure operations."
 plugins {
     `maven-publish`
     kotlin("multiplatform")
-    id("org.jetbrains.kotlin.plugin.allopen")
-    id("kotlinx.benchmark")
-}
-
-allOpen {
-    annotation("org.openjdk.jmh.annotations.State")
-    annotation("org.openjdk.jmh.annotations.BenchmarkMode")
-}
-
-sourceSets.create("benchmark")
```

Comment:
see this thread from when I was converting this build to mpp originally: https://github.com/michaelbull/kotlin-result/pull/18#discussion_r436006070

I recall running into issues trying to get the benchmark plugin to register any benchmark source when it existed inside the mpp project. 
Defining the "jvmBenchmark" source set alone wasn't enough. I believe I tried to create it as a custom source set inside of the mpp plugin: https://kotlinlang.org/docs/reference/mpp-dsl-reference.html#custom-source-sets. But that nothing was being registered as a target by the benchmark plugin. The multiplatform plugin turned out to provide a sort of workaround which may not be needed anymore. By calling `withJava()` inside the mpp `jvm` block, java code in the jvmBenchmark sourceset was magically picked up when registered to the benchmark plugin.

So you can definitely remove `withJava()` from this file, as well as `val jvmMain by getting` as that was only needed for the jvmBenchmark dependencies.

but perhaps some experimentation is needed to see what's the minimum needed to get this working for the new file